### PR TITLE
Workaround for clang4 lexical_cast<__int128> #14

### PR DESCRIPTION
**Pull request by zxcat**
Note about `9020039` magic number: https://github.com/GolosChain/cyberway.fc/blob/a748b74e2f912d97222691583c80f30e735f818e/src/variant.cpp#L20
AppleClang have it's own version numbering, and `9020039` represents "clang-902.0.39.1", which is part of llvm 5.0.2 ([source](https://en.wikipedia.org/wiki/Xcode#Latest_versions)). It can be not precise, will fix if meet problem version

### CODE REVIEW

File: src/variant.cpp [hunk 1]
```diff
@@ -16,8 +16,24 @@ namespace {
...
-{
+#ifdef __APPLE__
+#  if __apple_build_version__ < 9300000 || __clang_major__ == 4   // 9.3 is clang 5, not actually checked, is it works there
+#     define ABSENT__INT128_TYPEID
+namespace boost {
+template<typename A>
+inline __int128 lexical_cast_128(const A& arg) {
+   __int128 result = 0;
+   if (!boost::conversion::detail::try_lexical_convert(arg, result)) {
+      boost::throw_exception(bad_lexical_cast());
```
[the remark this answers is not in the document]
Comment (e-schepachev) [reply src/variant.cpp]: O no necessity. Message will be changed in the calling method

File: src/variant.cpp [hunk 2]
```diff
@@ -526,7 +542,11 @@ __uint128 variant::to_uint128() const {
     if( is_uint128() || is_int128()) {
         return value_.as_uint128;
     } else if (is_string()) {
+#ifndef ABSENT__INT128_TYPEID
        return boost::lexical_cast<__int128>(value_.as_string);
+#else
+       return boost::lexical_cast_128(value_.as_string);
+#endif
```
Comment (e-schepachev) [inline src/variant.cpp]: may be better to hide this inside boost::lexical_cast_128(value_.as_string);
And only call it there.

File: src/variant.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (zxcat) [reply src/variant.cpp]: `lexical_cast_128` is copy-pasted from boost implementation, the only difference is in throwing exception. if boost have type info (`BOOST_NO_TYPEID` not defined), it calls `boost::throw_exception(bad_lexical_cast(typeid(Source), typeid(Type)));`, if it haven't, it calls `boost::throw_exception(bad_lexical_cast());`. The problem is that type info enabled, but clang 4 (on macos?) have no type info for `__int128`

File: src/variant.cpp (hunk 2 above)
Comment (zxcat) [reply src/variant.cpp]: agree

**Pull request merged**